showcase/gapic-showcase/src/test/java/com/google/showcase/v1beta1/it/ITOtelMetrics.java

```diff
@@ -0,0 +1,554 @@
+    File file = new File(filePath);
+    Truth.assertThat(file.exists()).isTrue();
+    Truth.assertThat(file.length() > 0).isTrue();
+
+    client.close();
+    client.awaitTermination(TestClientInitializer.AWAIT_TERMINATION_SECONDS, TimeUnit.SECONDS);
+  }
+
+  @Test
+  public void testGrpc_OperationCancelled_recordsMetrics() throws Exception {
+
+    // initialize the otel-collector
+    setupOtelCollector(
+        "../scripts/start_otelcol.sh",
+        "../directory4",
+        "../opentelemetry-helper/configs/testGrpc_OperationCancelled.yaml");
+
+    EchoSettings grpcEchoSettings =
+        EchoSettings.newBuilder()
+            .setCredentialsProvider(NoCredentialsProvider.create())
+            .setTracerFactory(createOpenTelemetryTracerFactory("4320"))
+            .setTransportChannelProvider(
+                EchoSettings.defaultGrpcTransportProviderBuilder()
+                    .setChannelConfigurator(ManagedChannelBuilder::usePlaintext)
+                    .build())
+            .setEndpoint("localhost:7469")
+            .build();
+
+    EchoClient client = EchoClient.create(grpcEchoSettings);
+
+    EchoRequest requestWithNoError =
+        EchoRequest.newBuilder()
+            .setError(Status.newBuilder().setCode(Code.OK.ordinal()).build())
+            .build();
+
+    client.echoCallable().futureCall(requestWithNoError).cancel(true);
+
+    // wait for the metrics to get uploaded
+    Thread.sleep(5000);
+
+    String filePath = "../directory4/testGrpc_OperationCancelled_metrics.txt";
+    // 1. Check if the log file exists
+    File file = new File(filePath);
+    Truth.assertThat(file.exists()).isTrue();
+    Truth.assertThat(file.length() > 0).isTrue();
+
+    client.close();
+    client.awaitTermination(TestClientInitializer.AWAIT_TERMINATION_SECONDS, TimeUnit.SECONDS);
+  }
+
+  @Test
+  public void testHttpJson_OperationFailed_recordsMetrics() throws Exception {
+
+    // initialize the otel-collector
+    setupOtelCollector(
+        "../scripts/start_otelcol.sh",
+        "../directory5",
+        "../opentelemetry-helper/configs/testHttpJson_OperationFailed.yaml");
+
+    EchoSettings httpJsonEchoSettings =
+        EchoSettings.newHttpJsonBuilder()
+            .setCredentialsProvider(NoCredentialsProvider.create())
+            .setTracerFactory(createOpenTelemetryTracerFactory("4321"))
+            .setTransportChannelProvider(
+                EchoSettings.defaultHttpJsonTransportProviderBuilder()
+                    .setHttpTransport(
+                        new NetHttpTransport.Builder().doNotValidateCertificate().build())
+                    .setEndpoint("http://localhost:7469")
+                    .build())
+            .build();
+    EchoClient client = EchoClient.create(httpJsonEchoSettings);
+
+    EchoRequest requestWithError =
+        EchoRequest.newBuilder()
+            .setError(Status.newBuilder().setCode(Code.PERMISSION_DENIED.ordinal()).build())
+            .build();
```

Comment:
qq, also with this. Some tests are using PERMISSION_DENIED, AUTHENTICATED, and INVALID_ARGUMENT? Is there a reason for the different status code?

Reply:
There is no specific reason. I will use these status codes to verify the metrics collected, so wanted to use as many different codes as I can.